### PR TITLE
Clarify 'describe your project' question in Issue template.

### LINKED ISSUE #39: Features and Enhancements should be treated separately or questions be rephrased
Relates to #10.

Noticed some issue with people not filling templates properly.

There should be two kinds of templates in GIP: one handling new features, and another enhancing existing ones, that way people could avoid filling out redundant questions. The template itself is named as `Feature / Enhancement Proposal` when you attempt to create one, so makes sense to split those just like in the main repository.

I propose modifying those templates like the following:

# Feature template
~~**Describe the project  you are working on**~~
**Describe the vision/end goal you are trying to achieve**
* not sure how to phrase it better but some people might not want to describe their own projects for whatever reason (they even might be under NDA, who knows).

~~**Describe how this enhancement will help your project:**~~
**Describe how this feature will help to achieve your goal:**

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**

**Describe implementation detail for your proposal (in code), if possible:**

~~**If this feature will not be used often, can it be worked around with a few lines of script?:**~~
* this feels a bit more like a hint/guide rather than an actual question to fill.
* this also pertains to enhancement rather than feature proposal.
* "working around" makes sense for existing features.
* the feature should be used relatively often within domain!

**Is there a reason why this should be core and not an add-on in the asset library?:**

# Enhancement template
~~**Describe the project you are working on**~~
**Describe enhancement of the existing feature** (https://github.com/godotengine/godot-proposals/issues/39#issuecomment-528819736)
* most people refer to this as "irrelevant" when filling out blanks #37, #35.
* some just learn stuff #28.

~~**Describe how this feature / enhancement will help your project:**~~
**Describe how this enhancement will help to achieve your goal:**

~~**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**~~
* somewhat irrelevant because the ground work is already done.

**Describe implementation detail for your proposal (in code), if possible:**

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**

~~**Is there a reason why this should be core and not an add-on in the asset library?:**~~
* the feature is already there in the core #37, #36

---

A few other questions could be added for each.

### PR DESCRIPTION
**Pull request by willnationsdev**
cc @Xrayez @clayjohn 

Closes #39 so-to-speak. Questions won't be removed, but they can be reworded for clarity. The intent is to restrict proposals to only those that are relevant for actual work being done by the community (things that solve real obstacles / use-cases). The first question is meant to elicit an explanation of what that use-case is, put the proposal into context, and ward away people who are creating Issues just to request things they would *like* to have rather than *need* to have (as contributors do not have time/manpower to implement other features). It now more accurately states that the user must be trying to address a real problem in a real project, tool, or other type of task - and if they aren't, then they shouldn't be making a proposal.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature---enhacement-proposal.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ assignees: ''
 
 ---
 
-**Describe the project you are working on:**
+**You should have a live or in-development project/task/tool that you cannot complete as desired without this proposal. If so, please describe it:**
 
 **Describe how this feature / enhancement will help your project:**
```
Comment (Xrayez) [inline .github/ISSUE_TEMPLATE/feature---enhacement-proposal.md]: This needs `project/task/tool` as well.

**Comment by nobuyukinyuu** [comment]
Does this make the question asking if there could be a workaround redundant?  That's the question where I'd normally put project-relevant information.

**Comment by Xrayez** [comment]
I'm not sure now if there's a need to change this given the stance of core devs regarding the requirements ("what is needed" vs "what is wanted").

I'd much rather prefer this question to be removed completely to be honest. Most people just don't reveal what they are currently working on despite the question. If they do, it's a game genre information at most, or simply editor-specific context: "Top-down shooting game, but it doesn't matter, it's UI". This question currently present in the template is enough in my opinion:

**Describe the problem or limitation you are having in your project:**

---

Please don't force people to reveal their ideas. Ask them for use cases, not real-life projects. Godot was not open-source for a long time so I'd like @reduz to respect the choice of people not revealing what they are currently working on if there's no need for that. If a certain use case is not obvious without the context, simply close such a proposal, or ask the person to provide more concrete, real-life use cases. Otherwise it just signifies that the actual use cases are too specific to be implemented in core anyway, so I don't see the point.

If `godot-proposals` was created just to "filter out the noise", then I'm not sure whether the previous discussion was worth it.

@reduz probably hates me by now. :eyes:

Feel free to close this, sorry to bother you folks, I don't want to look hostile, I'm just learning to express my opinion (the most difficult thing for me to do, especially in a language with words and letters so distant from my ~~gd~~native). 🙂

**Comment by aaronfranke** [comment]
I don't really think this is an improvement. The existing question is a fairly short way to ask users to provide their use cases for the feature. This proposal also seems a bit strange since it asks for a project that cannot be completed without the proposal, yet I think we should allow proposals that simply improve or simplify the workflow.

So, I vote to close this.

**Comment by Xrayez** [comment]
> This proposal also seems a bit strange since it asks for a project that cannot be completed without the proposal

Yes, many existing proposals would have to be closed with this requirement. In fact this is also true for the current template. So currently it's either allowing people to put everything under "describe your project" without closing the proposal, or  there should be an editor-specific template (again this is also described in #39.

**Comment by aaronfranke** [comment]
Closing for all of the above reasons, plus there is a conflict anyway.

**Pull request closed without merging**